src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -964,6 +979,22 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             filterOpt: null,
                             cancellationToken: cancellationToken);
 
+                        // Prior to generating the xml documentation file,
+                        // we apply programmatic suppressions for compiler warnings from diagnostic suppressors.
+                        // If there are still any unsuppressed errors or warnings escalated to errors
+                        // then we bail out from generating the documentation file.
+                        // This maintains the compiler invariant that xml documentation file should not be
+                        // generated in presence of diagnostics that break the build.
+                        if (analyzerDriver != null && !diagnostics.IsEmptyWithoutResolution)
```

Comment:
Here you check `!diagnostics.IsEmptyWithoutResolution` but below you don't. Why the difference?